hub/models.py

```diff
@@ -1077,18 +1088,31 @@ def get_import_data(self):
         return GenericData.objects.filter(
             data_type__data_set__external_data_source_id=self.id
         )
+    
+    @cached_property
+    def analytics(self):
+        return Analytics(qs=self.get_import_data)
 
     def get_import_dataframe(self):
-        enrichment_data = self.get_import_data()
-        json_list = [d.json for d in enrichment_data]
-        enrichment_df = pd.DataFrame.from_records(json_list)
-        return enrichment_df
+        return self.analytics.get_dataframe()
+    
+    def imported_data_count_by_region(self) -> List[Analytics.RegionCount]:
+        return self.analytics.imported_data_count_by_region()
+    
+    def imported_data_count_by_constituency(self) -> List[Analytics.RegionCount]:
+        return self.analytics.imported_data_count_by_constituency()
+    
+    def imported_data_count_by_constituency_2024(self) -> List[Analytics.RegionCount]:
+        return self.analytics.imported_data_count_by_constituency_2024()
+    
+    def imported_data_count_by_council(self) -> List[Analytics.RegionCount]:
+        return self.analytics.imported_data_count_by_council()
+    
+    def imported_data_count_by_ward(self) -> List[Analytics.RegionCount]:
+        return self.analytics.imported_data_count_by_ward()
```

Comment:
Think this could be more elegantly done by subclassing the base class.

Reply:
Yeah, but I think this is a refactor when we touch this code again, rather than being a for now thing.